### DISCUSSION TITLE
Update

### DISCUSSION THREAD

**Discussion by Mickael81** [z-huang/music#1741] [General]
Hello,
The latest version of the application (v.0.5.10) dates back to September 1st. 
Why have new updates not been published since that date?
What are the reasons why they are not more regular?
Even if all the reported anomalies are not resolved, publishing a monthly update would seem more judicious to me. What about you?

**Comment by nubesurrealista** [comment]
This is a free and open source project. The dev is not our employee, as simple as that. There will be updates when he wants or can do it.

**Comment by Mickael81** [comment]
I understand that although this application is developed on free time. 

The latest version published works very well on my device. I myself presented my contribution to the project. (It may not be much but if it can help ...) 

On the other hand, I deplore that there is no feedback on these contributions! I do not even know if my contribution was taken into account or not and if it interests the developer! 

Personally, I find it disrespectful towards the contributors! 
Other developers have taken my contribution into account by leaving me a thank you message telling me that it will appear on the next update. It was put online in less than 2 weeks. 

For an identical project, I saw that there was very good follow-up behind it and regular updating. 
For comparison, InnerTune has not been updated since September and its future really leaves me perplexed ...

**Comment by Mickael81** [comment]
I don't want to play the pessimist but, in my opinion, the development of this application is falling into disuse... It's really a shame. Oh well, too bad, I'll contribute to other applications listed on GitHub. Some developers are a little more reactive.

**Comment by rcky844** [comment]
Hi,

I have compiled all the recent PRs together and made an unofficial release build v0.5.11.

Changes include:
- All changes after initial v0.5.10 release
- Merge #1789, #1808 and #1815
- Updated library dependencies
- Enabled support for Android versions 5.0+

You can find the APK (packaged in ZIP) here:
[app-foss-release.zip](https://github.com/user-attachments/files/19120561/app-foss-release.zip)

Source: https://github.com/rcky844/InnerTune

(Note: This APK is signed with my own keys, so you can't update directly from the official v0.5.10).

**Comment by rcky844** [comment]
Hi again,

I've made some more changes to address other issues, and all the deviations from the official v0.5.10 release are listed below:

- All changes on `dev` branch after initial v0.5.10 release
- Merge #1789, #1808 and #1815
- Updated library dependencies
- Enabled support for Android versions 5.0+
- #1817: Disabled zoom control buttons in YouTube music login page
- #1818: Fixes for Discord RPC

You can find the APK (packaged in ZIP) here:
[app-foss-release.zip](https://github.com/user-attachments/files/19130420/app-foss-release.zip)

Source: https://github.com/rcky844/InnerTune

(Note: This APK is signed with my own keys, so you can't update directly from the official v0.5.10).

**Reply by rcky844** [reply]
![image](https://github.com/user-attachments/assets/14d6a3af-7770-45d4-83f4-05425b0171c8)
Here's a screenshot of InnerTune functioning on Android Lollipop.

**Comment by Mickael81** [comment]
Hi,
@rcky844 Thank you very much for continuing to develop this project! 🙏 
I downloaded the application and the changes have been applied. 👍
Too bad you can't share the application directly on your page. 
I hope you can continue to make it more functional. 😉

**Comment by rcky844** [comment]
New update with some fixes for various things, mostly ported from OuterTune.

Changelog (so far):
- All changes on `dev` branch after the v0.5.10 release
- Merged #1789, #1808 and #1815
- Updated library dependencies
- Enabled support for Android versions 5.0+
- #1817: Disabled zoom control buttons in YouTube music login page
- #1818: Fixes for Discord RPC
- Added support for poTokens & dataSyncId: https://github.com/OuterTune/OuterTune/pull/294, https://github.com/OuterTune/OuterTune/pull/326
- Fixed issues with brand accounts: https://github.com/OuterTune/OuterTune/issues/27

Download:
[app-foss-release.zip](https://github.com/user-attachments/files/19201663/app-foss-release.zip)

**Comment by rcky844** [comment]
[app-foss-release.zip](https://github.com/user-attachments/files/19221221/app-foss-release.zip)

This is a quick update fixing lyrics crashes on Android Lollipop and Marshmallow.

**Comment by rcky844** [comment]
New update with some fixes with important fixes, notably playlist and browsing fixes.

Changelog (so far):
- All changes on `dev` branch after the v0.5.10 release
- Merged #1789, #1808 and #1815
- Updated library dependencies
- Enabled support for Android versions 5.0+
- #1817: Disabled zoom control buttons in YouTube music login page
- #1818: Fixes for Discord RPC
- #1822: Fixed issues for playlists with over 100 songs
- Added support for poTokens & dataSyncId: https://github.com/OuterTune/OuterTune/pull/294, https://github.com/OuterTune/OuterTune/pull/326
- Fixed issues with brand accounts: https://github.com/OuterTune/OuterTune/issues/27
- Added option to logout from YouTube Music account
- Updated YouTube client hints

Download:
[app-foss-release.zip](https://github.com/user-attachments/files/19268812/app-foss-release.zip)

**Comment by rcky844** [comment]
[app-foss-release.zip](https://github.com/user-attachments/files/19270075/app-foss-release.zip)

Excuse my sheer volume of updates, this fixes issues with some search queries (https://github.com/z-huang/InnerTune/pull/1823).

**Comment by rcky844** [comment]
[app-foss-release.zip](https://github.com/user-attachments/files/19288278/app-foss-release.zip)

Further enhanced fixes for search queries, space should work fine now. Also merged new PR for translation fixes.

**Comment by rcky844** [comment]
[app-foss-release.zip](https://github.com/user-attachments/files/19324843/app-foss-release.zip)

Proper fix for search queries, report if any bugs are found.

**Reply by internet-coffee** [reply]
the settings icon continued show red dot.Even though I have updated to the latest
![image](https://github.com/user-attachments/assets/431b4129-9988-4b17-89dc-883428384243)

**Reply by StrugglingOnion** [reply]
why cant i use my backup file from previous version

**Reply by rcky844** [reply]
> the settings icon continued show red dot.Even though I have updated to the latest

Yup, that's a known thing, and I don't think it is that big of an issue to look into fixing it for now, sorry!

> why cant i use my backup file from previous version

It should technically work? I have attempted that when I updated myself.

**Reply by dam00042** [reply]
I still cannot import over 100 songs playlists, is there any way to solve it that I am not considering? The number of songs appeared correctly but then if you go into the playlist or import it, it still has 99 or 100 songs. I attach images with the problem.
![photo1](https://github.com/user-attachments/assets/2ed40557-125b-4220-8b2b-1e1819269747)
![photo2](https://github.com/user-attachments/assets/0cf0185c-4cd9-43d9-aeea-b669ced3cd5c)

**Reply by d4f5409d** [reply]
### Hey Ricky!

We would like you to help us a bit in overlooking these changes in this InnerTune fork: https://github.com/Malopieds/InnerTune/issues/463#issuecomment-2745139232

We would appreciate if you could help us. Thank you for your work!

**Reply by rcky844** [reply]
> I still cannot import over 100 songs playlists, is there any way to solve it that I am not considering? The number of songs appeared correctly but then if you go into the playlist or import it, it still has 99 or 100 songs. I attach images with the problem.

The problem is acknowledged. I counted a playlist loaded online and it seems like the total number of entries rendered is only 100, so the issue of loading playlists with 100 tracks is fixed while the issue of tracking the 101th and beyond music is not.

Frankly, I have little prior experience in Jetpack Compose, and this app provided me with the first experience of it. For such a huge codebase like this, it takes time for me to process and find solutions to issues. Combined with the fact that I am busy with life atm, I cannot provide a proper fix for your issue right now.

I think a quick solution for anyone who's a developer would be to cherry-pick relevant changes from InnerTune forks, but that's way beyond my head right now.

> ### Hey Ricky!
> 
> We would like you to help us a bit in overlooking these changes in this InnerTune fork: [Malopieds#463 (comment)](https://github.com/Malopieds/InnerTune/issues/463#issuecomment-2745139232)
> 
> We would appreciate if you could help us. Thank you for your work!

Certainly, but sadly not now! I appreciate you informing me about this though.

**Reply by dam00042** [reply]
Ok, no problem, it's true that it is a huge codebase and it's not a simple thing to fix. I was just giving feedback just in case. Thank you for your response! I hope this application or any fork can be improved because for me, it has such a potential...

**Reply by RainPlays09** [reply]
> [app-foss-release.zip](https://github.com/user-attachments/files/19324843/app-foss-release.zip)
> 
> Proper fix for search queries, report if any bugs are found.

It throws an "unknown error" again!

![Screenshot_20250325-104611.jpg](https://github.com/user-attachments/assets/30aca367-636a-4de8-8686-3a53e0c96d76)

**Reply by rcky844** [reply]
> > [app-foss-release.zip](https://github.com/user-attachments/files/19324843/app-foss-release.zip)
> > Proper fix for search queries, report if any bugs are found.
> 
> It throws an "unknown error" again!

Please refer to https://github.com/OuterTune/OuterTune/issues/464

Edit: Current workaround is to wipe data and play music in offline mode.

**Comment by rcky844** [comment]
[app-foss-release.zip](https://github.com/user-attachments/files/19471360/app-foss-release.zip)

Update with playback fixes, should work fine with logging in now.

**Reply by SheepCommander** [reply]
thank you!

**Reply by SheepCommander** [reply]
It sometimes has Unknown Error after around 30 seconds per song :(

**Reply by ScimpanzeS** [reply]
Still doesn't work. Unknown error in every song with or without login

**Reply by zara-225** [reply]
I cant make playlists over 100 songs. When i import a playlist  from youtube it only adds  the last 100 songs for some reason. It also shows 
in the app on the youtube playlst that there are only  100 songs when there should be over 300

**Reply by shibotto** [reply]
Seems to work fine for me, although for now I only quickly tested it inside a VM:

https://github.com/user-attachments/assets/e009a13a-da11-4212-af62-ae0e5bbff051

Forgot to mention, NOT logged in.

**Reply by shibotto** [reply]
When importing a previous DB (if you come from Malopieds fork, [online tool to convert it](https://innertune.replit.app/)) I noticed I need a new visitor data key or I can't fetch anything. The easiest way to fix it is to delete `settings.preferences_pb` inside the backup zip file, which also means losing your appearance, region, and playback settings (**not** anything related to your library, likes, playlists, etc). But this app only has a handful of options and it's a <1 minute job to reconfigure it.

**Reply by internet-coffee** [reply]
any update?😭

**Reply by SheepCommander** [reply]
@kevin970712 if one fork still isn't working for you I'd try another fork, like OuterTune etc.

I'm currently using Metrolist since it has Android Auto support, works on both my Wifi and T Mobile almost all the time (I've only had one playback error in like 7 months)

**Reply by internet-coffee** [reply]
@SheepCommander I've been using Metrolist, but it's not as lightweight as Innertune. I'm hoping for a project that can maintain the underlying operation of Innertune.

**Reply by SheepCommander** [reply]
If you want a list of forks I know other than the two mentioned, there's also OpenTune and Muzza which are still semi active. Searching "YouTube Music" on F Droid gave a few more projects like Kreate and SimpMusic.

I'm not a developer so that's all I can help with. You might also be interested in just ReVanced yt music. Good luck!

**Reply by Mickael81** [reply]
Unfortunately, like many people, I fear that the InnerTune project is no longer being supported. It's a real shame because it had potential. 😔

If you're interested, you'll find a non-exhaustive list of applications in this thread that are well-maintained and regularly updated:
https://github.com/chayotic/Open-Source-Music-Streaming-Apps

**Reply by Dimaskon** [reply]
Please comeback, with update bro. We still waiting. I hope you Will be read this🥹
